networkdb/networkdb.go

```diff
@@ -61,6 +61,15 @@ type NetworkDB struct {
 	// for the network that node is participating in.
 	networks map[string]map[string]*network
 
+	// Per network epoch ID. Clients of networkdb can join & leave the
+	// with same network ID multiple times. Each instantiation of network
+	// in the networkdb gets its own Epoch ID. And the events associated
+	// with the network all get stamped with the current epoch ID.
+	// networkDB uses an eventual consistency mechanism to distribute the
+	// events across the cluster. Hence events can go out of order. Epoch ID
+	// helps the receiving nodes to discard older events
+	networkEpoch map[string]uint64
```

Comment:
can we track the network id of the current node under the network object in the nDB.networks[nDB.config.name]. This way the cleanup will come for free (disclaimer, today the cleanup of nDB.networks[nDB.config.name] is not done)